src/shared/codelens/csharpCodeLensProvider.ts

```diff
@@ -115,84 +123,96 @@ async function onLocalInvokeCommand({
-        'Preparing to run {0} locally...',
-        lambdaLocalInvokeParams.handlerName
-    )
-
     try {
-        if (!lambdaLocalInvokeParams.isDebug) {
-            const baseBuildDir = await makeBuildDir()
-            const templateDir = path.dirname(lambdaLocalInvokeParams.samTemplate.fsPath)
-            const documentUri = lambdaLocalInvokeParams.document.uri
-            const handlerName = lambdaLocalInvokeParams.handlerName
-
-            const handlerFileRelativePath = getHandlerRelativePath({
-                codeRoot: templateDir,
-                filePath: documentUri.fsPath
-            })
+        // Switch over to the output channel so the user has feedback that we're getting things ready
+        channelLogger.channel.show(true)
+        channelLogger.info(
+            'AWS.output.sam.local.start',
+            'Preparing to run {0} locally...',
+            lambdaLocalInvokeParams.handlerName
+        )
 
-            const relativeFunctionHandler = getRelativeFunctionHandler({
-                handlerName,
-                runtime,
-                handlerFileRelativePath
-            })
+        const baseBuildDir = await makeBuildDir()
+        const codeDir = path.dirname(lambdaLocalInvokeParams.document.uri.fsPath)
+        const documentUri = lambdaLocalInvokeParams.document.uri
+        const handlerName = lambdaLocalInvokeParams.handlerName
 
-            const lambdaInfo = await getLambdaInfoFromExistingTemplate({
-                workspaceUri: lambdaLocalInvokeParams.workspaceFolder.uri,
-                relativeOriginalFunctionHandler: relativeFunctionHandler
-            })
+        const inputTemplatePath = await makeInputTemplate({
+            baseBuildDir,
+            codeDir,
+            relativeFunctionHandler: handlerName,
+            runtime,
+        })
 
-            const properties = lambdaInfo ? lambdaInfo.resource.Properties : undefined
-            const codeDir = properties ? path.join(templateDir, properties.CodeUri) : templateDir
+        const buildArgs: ExecuteSamBuildArguments = {
+            baseBuildDir,
+            channelLogger,
+            codeDir,
+            inputTemplatePath,
+            samProcessInvoker: processInvoker,
+        }
+        if (lambdaLocalInvokeParams.isDebug) {
+            buildArgs.environmentVariables = {
+                SAM_BUILD_MODE: 'debug'
+            }
+        }
+        const samTemplatePath: string = await executeSamBuild(buildArgs)
+
+        const invokeArgs: InvokeLambdaFunctionArguments = {
+            baseBuildDir,
+            documentUri,
+            originalHandlerName: handlerName,
+            handlerName,
+            originalSamTemplatePath: inputTemplatePath,
+            samTemplatePath,
+            runtime,
+        }
 
-            const inputTemplatePath = await makeInputTemplate({
-                baseBuildDir,
-                codeDir,
-                relativeFunctionHandler,
-                properties,
```

Comment:
Properties contains things like envvars from the sam template. Why are they no longer used in making the input template?

Reply:
Good catch, fixed.
✔️